src/app/app/[username]/page.tsx

```diff
@@ -1,23 +1,27 @@
 import Profile from "@/components/Profile";
-import { authOptions } from "@/shared/auth";
-import { getServerSession } from "next-auth";
+import { prisma } from "@/shared/db";
+import { redirect } from "next/navigation";
 
 type Props = {
   params: {
     username: string;
   };
 };
 
-// FIXME: This error disabling should be removed as we use the username param in the future
-// eslint-disable-next-line no-unused-vars
 export default async function ProfilePage({ params: { username } }: Props) {
-  const session = await getServerSession(authOptions);
+  const profileUser = await prisma.user.findUnique({
+    where: { username: decodeURIComponent(username) },
+  });
+
+  if (!profileUser) {
+    redirect("/");
```

Comment:
```suggestion
    redirect(`/?error=${error}`);
```

Reply:
This doesn't seem right. I think you meant something different, right?

`export default async function ProfilePage({ params: { username } }: Props) {
  let profileUser = null;

  try {
    profileUser = await prisma.user.findUnique({
      where: { username: decodeURIComponent(username) },
    });
  } catch (error) {
    redirect(`/?error=${error}`);
  }

  return (
    <div className="min-h-full">
      <div className="bg-white shadow">
        <div className="flex items-center gap-4 mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
          <Profile profileUser={profileUser!} />
        </div>
      </div>
    </div>
  );
}`ts

Reply:
This doesn't seem right. I think you meant something different, right?

`export default async function ProfilePage({ params: { username } }: Props) {
  let profileUser = null;

  try {
    profileUser = await prisma.user.findUnique({
      where: { username: decodeURIComponent(username) },
    });
  } catch (error) {
    redirect(`/?error=${error}`);
  }

  return (
    <div className="min-h-full">
      <div className="bg-white shadow">
        <div className="flex items-center gap-4 mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
          <Profile profileUser={profileUser!} />
        </div>
      </div>
    </div>
  );
}`ts

Reply:
This doesn't seem right. You meant something else, right?

```
export default async function ProfilePage({ params: { username } }: Props) {
  let profileUser = null;

  try {
    profileUser = await prisma.user.findUnique({
      where: { username: decodeURIComponent(username) },
    });
  } catch (error) {
    redirect(`/?error=${error}`);
  }

  return (
    <div className="min-h-full">
      <div className="bg-white shadow">
        <div className="flex items-center gap-4 mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
          <Profile profileUser={profileUser!} />
        </div>
      </div>
    </div>
  );
}
```typescript

Reply:
Actually no. The error here is that the user was not found. In his case, we need to return something else.

I suggest:
1. create a function on `db.ts` to `findUserByUsername`, to avoid leaking its implementation.
```typescript
export function findUserByUsername(username) {
  return prisma.user.findUnique({
    where: { username: decodeURIComponent(username) },
  })
}
```
2. change this page to show something like
```typescript
export default async function ProfilePage({ params: { username } }: Props) {
  const profileUser = await findUserByUsername(username);

  return (
    <div className="min-h-full">
      <div className="bg-white shadow">
        <div className="flex items-center gap-4 mx-auto max-w-7xl px-4 py-6 sm:px-6 lg:px-8">
          {profileUser == null ? <ProfileNotFound /> : <Profile profileUser={profileUser!} />}
        </div>
      </div>
    </div>
  );
}

function ProfileNotFound() {
  // just an example, you can make something better than this
  return <p>Oops, this user was not found :(</p>
}
```